File: app/src/main/java/com/estrrado/tradopay/ui/modules/home/adapter/HomeItemsDetailAdapter.java
package com.estrrado.tradopay.ui.modules.home.adapter;

import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import com.estrrado.tradopay.ui.modules.OnItemSelectedlistner;
import com.example.tradopay.R;

public class HomeItemsDetailAdapter extends RecyclerView.Adapter<HomeItemsDetailAdapter.HomeItemsListViewHolder> {

    OnItemSelectedlistner onItemSelectedlistner;
    String[] arr;
    int[] imgs;
    public HomeItemsDetailAdapter(OnItemSelectedlistner onItemSelectedlistner,String[] arr,int[] imgs) {
        this.onItemSelectedlistner=onItemSelectedlistner;
        this.arr=arr;
        this.imgs=imgs;
    }

    @NonNull
    @Override
    public HomeItemsListViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        return new HomeItemsListViewHolder(LayoutInflater.from(parent.getContext()).inflate(R.layout.layout_home_detail_item,parent,false));
    }

    @Override
    public void onBindViewHolder(@NonNull HomeItemsListViewHolder holder, int position) {
        holder.img.setImageResource(imgs[position]);
        holder.txt.setText(arr[position]);
        holder.itemView.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                onItemSelectedlistner.onItemClick(null,position);
            }
        });

    }

    @Override
    public int getItemCount() {
        return imgs.length;
    }

    public class HomeItemsListViewHolder extends RecyclerView.ViewHolder
    {
        TextView txt;
        ImageView img;
        public HomeItemsListViewHolder(@NonNull View itemView) {
            super(itemView);
            txt=itemView.findViewById(R.id.txtItemName);
            img=itemView.findViewById(R.id.imgItem);
        }
    }
}

File: app/src/main/java/com/estrrado/tradopay/adapter/HomeItemsDetailAdapter.kt
package com.estrrado.tradopay.adapter

import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ImageView
import android.widget.TextView
import androidx.recyclerview.widget.RecyclerView
import com.estrrado.tradopay.listener.OnItemSelectedlistner
import com.example.tradopay.R

class HomeItemsDetailAdapter(var onItemSelectedlistner: OnItemSelectedlistner, var arr: Array<String>, var imgs: IntArray) : RecyclerView.Adapter<HomeItemsDetailAdapter.HomeItemsListViewHolder>() {
    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): HomeItemsListViewHolder {
        return HomeItemsListViewHolder(LayoutInflater.from(parent.context).inflate(R.layout.layout_home_detail_item, parent, false))
    }

    override fun onBindViewHolder(holder: HomeItemsListViewHolder, position: Int) {
        holder.img.setImageResource(imgs[position])
        holder.txt.text = arr[position]
        holder.itemView.setOnClickListener { onItemSelectedlistner.onItemClick(null, position) }
    }

    override fun getItemCount(): Int {
        return imgs.size
    }

    inner class HomeItemsListViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
        var txt: TextView
        var img: ImageView

        init {
            txt = itemView.findViewById(R.id.txtItemName)
            img = itemView.findViewById(R.id.imgItem)
        }
    }
}